src/pages/costModels/costModel/updateRateModel.test.tsx

```diff
@@ -259,127 +257,131 @@ describe('update-rate', () => {
   });
 
   test('submit regular', () => {
-    const { getByDisplayValue, getByText } = render(<RenderFormDataUI index={0} />);
-    fireEvent.change(getByDisplayValue(/openshift-aws-node/i), { target: { value: 'a new description' } });
-    expect(getByText(regExp(messages.Save)).closest('button').disabled).toBeFalsy();
-    fireEvent.click(getByText(regExp(messages.Save)));
+    render(<RenderFormDataUI index={0} />);
+    userEvent.type(screen.getByDisplayValue(/openshift-aws-node/i), 'a new description');
+    // eslint-disable-next-line testing-library/prefer-presence-queries
+    expect(screen.getByText(regExp(messages.Save)).getAttribute('disabled')).toBeNull();
+    userEvent.click(screen.getByText(regExp(messages.Save)));
   });
 
   test('regular', async () => {
-    const { getByLabelText, getByDisplayValue, getByText, getAllByRole } = render(<RenderFormDataUI index={0} />);
-    fireEvent.change(getByDisplayValue(/openshift-aws-node/i), { target: { value: 'a new description' } });
-    expect(getByText(regExp(messages.Save)).closest('button').disabled).toBeFalsy();
-    fireEvent.change(getByDisplayValue(/a new description/i), { target: { value: 'openshift-aws-node' } });
-    expect(getByText(regExp(messages.Save)).closest('button').disabled).toBeTruthy();
+    render(<RenderFormDataUI index={0} />);
+    const descInput = screen.getByDisplayValue('openshift-aws-node');
+    const saveButton = screen.getByText(regExp(messages.Save));
+    expect(saveButton.getAttribute('disabled')).not.toBeNull();
+    userEvent.clear(descInput);
+    userEvent.type(descInput, 'a new description');
+    expect(saveButton.getAttribute('disabled')).toBeNull();
+    userEvent.clear(descInput);
+    userEvent.type(descInput, 'openshift-aws-node');
+    expect(saveButton.getAttribute('disabled')).not.toBeNull();
 
     await waitFor(() => {
-      userEvent.click(document.querySelector(qr.measurement));
+      userEvent.click(screen.getByLabelText('measurement label'));
     });
-    userEvent.click(getAllByRole('option')[1]);
-    expect(getByText(regExp(messages.Save)).closest('button').disabled).toBeFalsy();
+    userEvent.click(screen.getAllByRole('option')[1]);
+    expect(saveButton.getAttribute('disabled')).toBeNull();
 
     await waitFor(() => {
-      userEvent.click(document.querySelector(qr.measurement));
+      userEvent.click(screen.getByLabelText('measurement label'));
```

Comment:
Should these strings come from our messages? I see a mix of both below

Reply:
testing-library does not like selectors that come from the deconstructed render. i added the label as a way to select the element, i'm open to other options though.

Reply:
I don't feel strongly about it, but would prefer if the strings came from messages. Ideally, strings would be maintained in one place.

Reply:
i discovered a `toggleAriaLabel` prop in the PF component that will allow me to put the aria label directly on the Select button, so I can target by that rather than having to deal with hidden label/labelby or testId+querySelector... I've also used the messages for strings.